### PR TITLE
Ft create test api db 162123750

### PR DESCRIPTION
**Pull request by bobbylemm**
What Does This PR do?
have tests written for the API endpoints and files renamed to reflect best practices.

What are The Relevant PT stories ID?
#162123750

### CODE REVIEW

File: Server/config/dbConfig.js [hunk 1]
```diff
@@ -18,12 +18,13 @@ const databaseConfig = {
...
-    host: process.env.DATABASE_URL,
-    port: process.env.DATABASE_URL,
-    database: process.env.DATABASE_URL,
-    password: process.env.DATABASE_URL
-  }
+  "production": process.env.DATABASE_URL
+  // {
+  //   user: process.env.DATABASE_URL,
+  //   host: process.env.DATABASE_URL,
+  //   port: process.env.DATABASE_URL,
```
Comment (saheedt) [inline Server/config/dbConfig.js:25]: please remove all unused code such as these

File: Server/controllers/ParcelController.js [hunk 2]
```diff
@@ -30,49 +30,64 @@ class ParcelController {
...
     if (!newStatus || !validStatus) {
         return res.status(400).json({
             message: 'please enter a valid status, as this is not valid'
         })
     }
     try {
-        const response = await parcelmanger.updateParcelStatus(newStatus, pid);
-        console.log(response);
+        await parcelmanger.updateParcelStatus(newStatus, pid);
+        // console.log(response);
```
Comment (saheedt) [inline Server/controllers/ParcelController.js:90]: Please remove all unused log statements too

File: Server/dataMigrations/createTables.js [hunk 3]
```diff
@@ -4,7 +4,13 @@ import config from '../config/dbConfig';
 // psql -U dozic -h127.0.0.1 senditdb
 
 dotenv.config();
-const pool = new pg.Pool(config.development);
+let configString = '';
+    if(process.env.NODE_ENV.trim() !== 'test') {
+        configString = config.development;
```
Comment (saheedt) [inline Server/dataMigrations/createTables.js:9]: With this statement here, you will never get to call the production database and in a case where you aren't in a test environment but the development database isn't available, you will always get an error.
Long story short, you are not catering for production database.

File: Server/dbManager/DbManager.js [hunk 4]
```diff
@@ -3,34 +3,38 @@ import dotenv from 'dotenv';
...
 
 class DbManager {
-  constructor() {
-    this.pool = new Pool(config.development);
-  }
+    constructor() {
+        let configString = '';
+        if(process.env.NODE_ENV.trim() !== 'test') {
+        configString = config.development;
+        } else {
```
Comment (saheedt) [inline Server/dbManager/DbManager.js:12]: you will also have the bug where you will never be able to use your production database

File: Server/dbManager/DbManager.js [hunk 5]
```diff
@@ -101,10 +99,19 @@ class DbManager {
     try {
         const q = 'SELECT packagename, dropofflocation, pickuplocation, price, presentlocation, weight, price, status FROM parcels;';
         const response = await this.pool.query(q);
-        // console.log(response);
```
Comment (saheedt) [inline Server/dbManager/DbManager.js:104]: please remove all unused code.

File: Server/test/route.test.js [hunk 6]
```diff
@@ -6,160 +6,221 @@ import app from '../app';
...
+                    expect(res.body.error.message).to.equal('you are not authorized to perform this action, admin only');
+                    expect(res.body).to.be.have.property('error');
+                    done();
+                })
+            })
+        })
+        describe("put/parcels/1/newlocation", () => {
+            it('should update a parcel presentlocation', (done) => {
+                chai.request(app)
+                .put('/api/v1/parcels/1/newlocation')
```
Comment (saheedt) [inline Server/test/route.test.js:195]: How did you come about this route? I'm pretty sure route for changing parcel location is `/parcels/<parcelId>/presentLocation`.
Did you even read the rubric or you just felt you needed your own endpoints?
You need to pay more attention to details.

File: Server/test/route.test.js [hunk 7]
```diff
@@ -6,160 +6,221 @@ import app from '../app';
...
+                    expect(res.body.error.message).to.equal('you are not authorized to perform this action, admin only');
+                    expect(res.body).to.be.have.property('error');
+                    done();
+                })
+            })
+        })
+        describe("put/parcels/newdropoff", () => {
+            it('should update a parcel dropoff location', (done) => {
+                chai.request(app)
+                .put('/api/v1/parcels/1/newdropoff')
```
Comment (saheedt) [inline Server/test/route.test.js:146]: this route i'm very sure should be ` /parcels/<parcelId>/destination`

**Review by saheedt — CHANGES_REQUESTED** [review]

File: Server/config/dbConfig.js (hunk 1 above)
Comment (bobbylemm) [reply Server/config/dbConfig.js:25]: i have taken them out

File: Server/controllers/ParcelController.js (hunk 2 above)
Comment (bobbylemm) [reply Server/controllers/ParcelController.js:90]: i have removed them all

File: Server/dataMigrations/createTables.js (hunk 3 above)
Comment (bobbylemm) [reply Server/dataMigrations/createTables.js:9]: i have made some adjustments to these

File: Server/dbManager/DbManager.js (hunk 5 above)
Comment (bobbylemm) [reply Server/dbManager/DbManager.js:104]: i have removed them all

File: Server/test/route.test.js (hunk 6 above)
Comment (bobbylemm) [reply Server/test/route.test.js:195]: i have renamed them

**Pull request closed without merging**